### ISSUE TITLE
Can't work customer Image attribute programmatically

### ISSUE THREAD

**Issue by cygnetampatel** [magento/magento2#19983]
### Preconditions (*)

1. Magento CE version 2.2.7
2. Create custom attribute programmatically, which is to be added to the 'customer eav attribute e.g.:
```
$customerSetup->addAttribute(\Magento\Customer\Model\Customer::ENTITY, "customerprofile",  array(
                "type"      => "varchar",
                "backend"   => "",
                "label"     => "Customerprofile",
                "input"     => "file",
                "source"    => "",
                "visible"   => true,
                "required"  => false,
                "default"   => "",
                "frontend"  => "",
                "unique"    => false,
                "note"      => ""
            ));
$customerdoc     = $customerSetup->getAttribute(\Magento\Customer\Model\Customer::ENTITY, "customerdoc");
```

```
$customerdoc     = $customerSetup->getEavConfig()->getAttribute(\Magento\Customer\Model\Customer::ENTITY, 'customerdoc');
            $used_in_forms[] = "adminhtml_customer";
            $used_in_forms[] = "checkout_register";
            $used_in_forms[] = "customer_account_create";
            $used_in_forms[] = "customer_account_edit";
            $used_in_forms[] = "adminhtml_checkout";
            $customerdoc->setData("used_in_forms", $used_in_forms)
                ->setData("is_used_for_customer_segment", true)
                ->setData("is_system", 0)
                ->setData("is_user_defined", 1)
                ->setData("is_visible", 1)
                ->setData("sort_order", 100);
            $customerdoc->save();

```
### Steps to reproduce (*)

1. Go to the registration form. you will show Customer profile attribute as a file. when user upload an image error message show "**Customer profile is not a valid file**".

### Expected result (*)

1. file is valid so data want to save.

### Actual result (*)

1. is not a valid file message on the frontend and backend.

### Issue In this file (*)
1. An issue in the if a condition  "Magento\Eav\Model\Attribute\Data\File"

    protected function _validateByRules($value)
    {
        if (!empty($value['tmp_name']) && !is_uploaded_file($value['tmp_name'])) {
            return [__('"%1" is not a valid file.', $label)];
        }
        return [];
    }
Acctually Image first of upload in temp folder then after move into accordinglly.

according to the condition if a file exists in temp folder it's throw error.

### Resolve the issue(*)

protected function _validateByRules($value)
    {
        if (empty($value['tmp_name']) && is_uploaded_file($value['tmp_name'])) {
            return [__('"%1" is not a valid file.', $label)];
        }
        return [];
    }

**Comment by magento-engcom-team** [comment]
Hi @cygnetampatel. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@cygnetampatel do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97352` were created

**Comment by magento-engcom-team** [comment]
Hi @Nazar65. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @cygnetampatel. Thank you for your report. 
The issue has been fixed in magento/magento2#19988 by @Nazar65 in 2.3-develop branch
Related commit(s):
 - [6d5ee6380cd9038fc1ef3403031e550f44440524](https://github.com/magento/magento2/commit/6d5ee6380cd9038fc1ef3403031e550f44440524)
 - [188a3e66e7d9604942d7c36cfaec7dc5bae32710](https://github.com/magento/magento2/commit/188a3e66e7d9604942d7c36cfaec7dc5bae32710)
 - [20fd58914382220cb4b5050d0540ec8f62c63814](https://github.com/magento/magento2/commit/20fd58914382220cb4b5050d0540ec8f62c63814)
 - [9ca8077242bd6bb7a09c8bb77f4495b39787fcb0](https://github.com/magento/magento2/commit/9ca8077242bd6bb7a09c8bb77f4495b39787fcb0)
 - [a0516563c682a9d3b51ffd0f0aa7040591dedf0a](https://github.com/magento/magento2/commit/a0516563c682a9d3b51ffd0f0aa7040591dedf0a)
 - [13722031d11aeb6ce92f0272db06714bce7840e0](https://github.com/magento/magento2/commit/13722031d11aeb6ce92f0272db06714bce7840e0)
 - [acc98eccca38b0915ab4edbe4606d254b9eb5cc3](https://github.com/magento/magento2/commit/acc98eccca38b0915ab4edbe4606d254b9eb5cc3)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @cygnetampatel. Thank you for your report. 
The issue has been fixed in magento/magento2#21437 by @Nazar65 in 2.2-develop branch
Related commit(s):
 - [b635e090921616459f254b63ae7de80f51a9e228](https://github.com/magento/magento2/commit/b635e090921616459f254b63ae7de80f51a9e228)
 - [97b8f5f60c4e970988ddbf3737740cdfae3054f8](https://github.com/magento/magento2/commit/97b8f5f60c4e970988ddbf3737740cdfae3054f8)

The fix will be available with the upcoming 2.2.9 release.
